cpp/core/jni/JniWrapper.cc

```diff
@@ -872,10 +871,9 @@ JNIEXPORT jlong JNICALL Java_io_glutenproject_vectorized_ShuffleWriterJniWrapper
     std::string errorMessage = "Invalid shuffle writer id " + std::to_string(shuffleWriterId);
     throw gluten::GlutenException(errorMessage);
   }
-  jlong evictedSize;
-  gluten::arrowAssertOkOrThrow(
-      shuffleWriter->evictFixedSize(size, &evictedSize), "(shuffle) nativeEvict: evict failed");
-  return evictedSize;
+  int64_t* evictedSize = nullptr;
+  gluten::arrowAssertOkOrThrow(shuffleWriter->evictFixedSize(size, evictedSize), "(shuffle) nativeEvict: evict failed");
+  return reinterpret_cast<jlong>(evictedSize);
```

Comment:
Does this safe?  seems before return jlong, after return int64_t pointer.

Reply:
this is a good point, fixed it